$OpenBSD: patch-libgnomevfs_gnome-vfs-ssl_c,v 1.1 2021/06/22 13:56:38 tb Exp $

Use an accessor instead of reaching inside the SSL struct.

Index: libgnomevfs/gnome-vfs-ssl.c
--- libgnomevfs/gnome-vfs-ssl.c.orig
+++ libgnomevfs/gnome-vfs-ssl.c
@@ -400,8 +400,7 @@ gnome_vfs_ssl_create_from_fd (GnomeVFSSSL **handle_ret
 			}
 		}
 
-                if (ssl->private->ssl->ctx)
-                        SSL_CTX_free (ssl->private->ssl->ctx);
+                SSL_CTX_free(SSL_get_SSL_CTX(ssl->private->ssl));
 
                 SSL_free (ssl->private->ssl);
 		g_free (ssl->private);
@@ -705,7 +704,7 @@ gnome_vfs_ssl_destroy (GnomeVFSSSL *ssl,
 		}
 	}
 	
-	SSL_CTX_free (ssl->private->ssl->ctx);
+	SSL_CTX_free (SSL_get_SSL_CTX(ssl->private->ssl));
 	SSL_free (ssl->private->ssl);
 	close (ssl->private->sockfd);
 	if (ssl->private->timeout)
